### PR TITLE
Implement the Ceremony/Bootstrap in the FT

### LINKED ISSUE #114: [Functional Tests] Add full step for the ceremony process using CLI
### What do you want to share with us?

Implement the functional tests using all the ceremony process using CLI
This will avoid releases with issues such as:
- https://github.com/vmware/repository-service-tuf-cli/issues/132

### References

- https://github.com/vmware/repository-service-tuf-cli/issues/132

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kairoaraujo**
Implement the Ceremony process and the Bootstrap in the functional tests.

The bootstrap process is done as a part of the RSTUF Service Setup. However, the steps are done once before running the BDD tests.

The PR adds the following steps in the workflow:

1. Deploy RSTUF Services (deploy the containers)
2. Install the rstuf CLI
3. [New] Run the rstuf-cli unit test to perform the _Ceremony_ 'Offline', which generates a fresh payload
  It guarantees that we are building a fresh `payload.json`, using the chosen rstuf CLI version. It avoids using static `payload.json` and tests the compatibility between rstuf-CLI and rstuf-API
4. [New] Bootstrap the RSTUF deployed in step 1, using the `payload.json` generated in step 3
5. Run Functional Tests -- BDD tests will always assume a Deployed RSTUF Service is already bootstrapped. 

Some notes:
- Step 3: An online Ceremony 'Online' is just a combination of steps 3 + 4. We make it separate for clarity/debugging.
- Step 4: We must assume that the functional tests always have a ready setup and ceremony/bootstrap is part of this setup
- This information will also be detailed in #52 

Minor fixes/enhancements:
- Moved to the GHA actions/checkout (`fetch-depth: 0`)
- Clean up white trailing spaces
- Added pretend dependency (required by unit tests)

Closes #114 

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

### CODE REVIEW

**Comment by kairoaraujo** [comment]
References for this review:
- Using the broken version of CLI (v0.0.1a5)
   https://github.com/vmware/repository-service-tuf/actions/runs/3394164269

- Using fixed CLI version (v0.0.1a6)
   https://github.com/vmware/repository-service-tuf/actions/runs/3394181573

**Comment by kairoaraujo** [comment]
I updated the PR details.

**Review by MVrachev — COMMENTED** [review]
Thanks for updating the pr details.
It's much better.

I have a couple of small suggestions.

File: .github/workflows/functional.yml
Comment (kairoaraujo) [reply .github/workflows/functional.yml:118]: IMHO, I don't think it is necessary to re-document a specific parameter from a GHA.
The [actions/checkout](https://github.com/actions/checkout#usage) already documents it.

File: .github/workflows/functional.yml
Comment (kairoaraujo) [reply .github/workflows/functional.yml:120]: We fetch all branches/tags, but it stays in the main.
`dev` -> main branch
`latest` -> latest release tag vX.Y.Z

File: requirements.txt
Comment (kairoaraujo) [reply requirements.txt:4]: I updated the PR details and added the `pretend` dependency and the reason.

Remember, we use pinned versions in `Pipfile` only when required.
`pipenv` command will constantly update the dependencies when necessary and pin them in the `Pipenv.lock`-- unless it is a pinned version in `Pipfile` to be reproducible.
We also use `pipenv` to generate the `requirements.txt` from `Pipenv.lock`

File: .github/workflows/functional.yml
Comment (MVrachev) [reply .github/workflows/functional.yml:120]: Ahaa, yes, correct. It makes sense.

**Review by MVrachev — APPROVED** [review]
The changes make sense.

Maybe it will be worth waiting for @rdimitrov approval as well as those changes are connected with the functional and BDD tests and will be good for him to read the changes as well.

**Review by rdimitrov — APPROVED** [review]

**Pull request merged**
